lightning-invoice/src/utils.rs

```diff
@@ -559,20 +569,27 @@ fn filter_channels<L: Deref>(
 		match filtered_channels.entry(channel.counterparty.node_id) {
 			hash_map::Entry::Occupied(mut entry) => {
 				let current_max_capacity = entry.get().inbound_capacity_msat;
-				if channel.inbound_capacity_msat < current_max_capacity {
+				let new_now_public = channel.is_public && !entry.get().is_public;
+				let new_higher_capacity = channel.is_public == entry.get().is_public &&
+					channel.inbound_capacity_msat > current_max_capacity;
```

Comment:
I had to draw a truth table to understand what these do, could you add some comments?